pop-api/src/v0/nfts.rs

```diff
@@ -18,14 +35,370 @@ pub fn mint(
+	/// Approve an item to be transferred by a delegated third-party account.
+	pub fn approve_transfer(
+		collection: CollectionId,
+		item: ItemId,
+		delegate: impl Into<MultiAddress<AccountId, ()>>,
+		maybe_deadline: Option<BlockNumber>,
+	) -> Result<()> {
+		Ok(dispatch(RuntimeCall::Nfts(NftCalls::ApproveTransfer {
+			collection,
+			item,
+			delegate: delegate.into(),
+			maybe_deadline,
+		}))?)
+	}
+
+	/// Cancel one of the transfer approvals for a specific item.
+	pub fn cancel_approval(
+		collection: CollectionId,
+		item: ItemId,
+		delegate: impl Into<MultiAddress<AccountId, ()>>,
+	) -> Result<()> {
+		Ok(dispatch(RuntimeCall::Nfts(NftCalls::CancelApproval {
+			collection,
+			item,
+			delegate: delegate.into(),
+		}))?)
+	}
+
+	/// Cancel all the approvals of a specific item.
+	pub fn clear_all_transfer_approvals(collection: CollectionId, item: ItemId) -> Result<()> {
+		Ok(dispatch(RuntimeCall::Nfts(NftCalls::ClearAllTransferApprovals { collection, item }))?)
+	}
+}
+
+pub mod attributes {
+	use super::*;
+
+	/// Approve item's attributes to be changed by a delegated third-party account.
+	pub fn approve_item_attribute(
+		collection: CollectionId,
+		item: ItemId,
+		delegate: impl Into<MultiAddress<AccountId, ()>>,
+	) -> Result<()> {
+		Ok(dispatch(RuntimeCall::Nfts(NftCalls::ApproveItemAttribute {
+			collection,
+			item,
+			delegate: delegate.into(),
+		}))?)
+	}
+
+	/// Cancel the previously provided approval to change item's attributes.
+	pub fn cancel_item_attributes_approval(
+		collection: CollectionId,
+		item: ItemId,
+		delegate: impl Into<MultiAddress<AccountId, ()>>,
+	) -> Result<()> {
+		Ok(dispatch(RuntimeCall::Nfts(NftCalls::CancelItemAttributesApproval {
+			collection,
+			item,
+			delegate: delegate.into(),
+		}))?)
+	}
+
+	/// Set an attribute for a collection or item.
+	pub fn set_attribute(
+		collection: CollectionId,
+		maybe_item: Option<ItemId>,
+		namespace: AttributeNamespace<AccountId>,
+		key: BoundedVec<u8, KeyLimit>,
+		value: BoundedVec<u8, KeyLimit>,
+	) -> Result<()> {
+		Ok(dispatch(RuntimeCall::Nfts(NftCalls::SetAttribute {
+			collection,
+			maybe_item,
+			namespace,
+			key,
+			value,
+		}))?)
+	}
+
+	/// Clear an attribute for a collection or item.
+	pub fn clear_attribute(
+		collection: CollectionId,
+		maybe_item: Option<ItemId>,
+		namespace: AttributeNamespace<AccountId>,
+		key: BoundedVec<u8, KeyLimit>,
+	) -> Result<()> {
+		Ok(dispatch(RuntimeCall::Nfts(NftCalls::ClearAttribute {
+			collection,
+			maybe_item,
+			namespace,
+			key,
+		}))?)
+	}
+
+	/// Get the attribute value of `item` of `collection` corresponding to `key`.
+	pub fn attribute(
+		collection: CollectionId,
+		item: ItemId,
+		key: BoundedVec<u8, KeyLimit>,
+	) -> Result<Option<Vec<u8>>> {
+		Ok(state::read(RuntimeStateKeys::Nfts(NftsKeys::Attribute(collection, item, key)))?)
+	}
+
+	// /// Get the custom attribute value of `item` of `collection` corresponding to `key`.
+	// pub fn custom_attribute(
```

Comment:
Why is this method commented out?

Reply:
As above, see comment in description about AccountId issues.